Formula/ice.rb

```diff
@@ -1,134 +1,89 @@
+      "LMDB_HOME=#{Formula["lmdb"].opt_prefix}",
+      "CONFIGS=shared cpp11-shared #{build.with?("xcode-sdk") ? "xcodesdk cpp11-xcodesdk" : ""}",
+      "PLATFORMS=all",
+      "SKIP=slice2confluence #{build.without?("additional-compilers") ? "slice2py slice2rb slice2js" : ""}",
+      "LANGUAGES=cpp objective-c #{build.with?("java") ? "java java-compat" : ""}",
     ]
-
-    cd "cpp" do
-      system "make", "install", *args
-    end
-
-    cd "objective-c" do
-      system "make", "install", *args
-    end
-
-    if build.with? "java"
-      cd "java" do
-        system "make", "install", *args
-      end
-    end
-
-    cd "php" do
-      args << "install_phpdir=#{share}/php"
-      args << "install_libdir=#{lib}/php/extensions"
-      system "make", "install", *args
-    end
+    system "make", "install", *args
   end
 
   test do
-    (testpath/"Hello.ice").write <<-EOS.undent
-      module Test {
-        interface Hello {
-          void sayHello();
-        };
-      };
+    (testpath / "Hello.ice").write <<-EOS.undent
+      module Test
+      {
+          interface Hello
+          {
+              void sayHello();
+          }
+      }
     EOS
-    (testpath/"Test.cpp").write <<-EOS.undent
+    (testpath / "Test.cpp").write <<-EOS.undent
       #include <Ice/Ice.h>
       #include <Hello.h>
 
-      class HelloI : public Test::Hello {
+      class HelloI : public Test::Hello
+      {
       public:
-        virtual void sayHello(const Ice::Current&) {}
+        virtual void sayHello(const Ice::Current&) override {}
       };
 
-      int main(int argc, char* argv[]) {
-        Ice::CommunicatorPtr communicator;
-        communicator = Ice::initialize(argc, argv);
-        Ice::ObjectAdapterPtr adapter =
-            communicator->createObjectAdapterWithEndpoints("Hello", "default -h localhost -p 10000");
-        adapter->add(new HelloI, communicator->stringToIdentity("hello"));
+      int main(int argc, char* argv[])
+      {
+        Ice::CommunicatorHolder ich(argc, argv);
+        auto adapter = ich->createObjectAdapterWithEndpoints("Hello", "default -h localhost -p 10000");
+        adapter->add(std::make_shared<HelloI>(), Ice::stringToIdentity("hello"));
         adapter->activate();
-        communicator->destroy();
         return 0;
       }
     EOS
     system "#{bin}/slice2cpp", "Hello.ice"
-    system "xcrun", "clang++", "-c", "-I#{include}", "-I.", "Hello.cpp"
-    system "xcrun", "clang++", "-c", "-I#{include}", "-I.", "Test.cpp"
-    system "xcrun", "clang++", "-L#{lib}", "-o", "test", "Test.o", "Hello.o", "-lIce", "-lIceUtil"
-    system "./test", "--Ice.InitPlugins=0"
-    system "/usr/bin/php", "-d", "extension_dir=#{lib}/php/extensions",
-                           "-d", "extension=IcePHP.dy",
-                           "-r", "extension_loaded('ice') ? exit(0) : exit(1);"
+    system "xcrun", "clang++", "-DICE_CPP11_MAPPING", "-std=c++11", "-c", "-I#{include}", "-I.", "Hello.cpp"
```

Comment:
@ilovezfs, or is the issue with using compilers in general inside the test block?